openqa-investigate

```diff
@@ -83,6 +93,34 @@ clone() {
     done
 }
 
+get_worker_settings() {
+    declare -n worker=$1
+    local ret
+    worker_vars_json=$(echo "$vars_json" | runjq -r '.WORKER_CLASS') || return $?
+    # Nothing to do if worker_class is not defined
+    if [[ $worker_vars_json == null ]]; then
+        ret=$?
+    fi
+    # get worker_class from settings to compare with vars.json
+    workerclass_in_settings=$(echo "$clone_job_data" | runjq -r '.job.settings.WORKER_CLASS') || return $?
+    # if workers from settings has more than one worker iterate and verify existence
+    if (echo "${workerclass_in_settings}" | grep -q ','); then
+        mapfile -td ',' workers_settings < <(echo "${workerclass_in_settings}")
+        mapfile -td ',' workers < <(echo "${worker_vars_json}")
+        for worker_item in "${workers_settings[@]}"; do
+            # check if worker is in the vars.json
+            if [[ "${workers[*]}" =~ $worker_item ]]; then
+                worker+=" ${worker_item}" && ret=$?
+            else
+                worker+=" unidentified worker ${worker_item}" && ret=$?
+            fi
+        done
+    else
+        worker="${workerclass_in_settings}" && ret=$?
+    fi
```

Comment:
As discussed just simplify to read the WORKER_CLASS from vars.json from the original job, don't care about the settings table.